lib/versioning/VersionID.js

```diff
@@ -152,4 +156,73 @@ function decode(str) {
     }
 }
 
-module.exports = { generateVersionId, getInfVid, encode, decode };
+/**
+ * Encode a versionId to obscure internal information contained
+ * in a version ID (less than 40 bytes).
+ *
+ * @param {string} str - the versionId to encode
+ * @return {string} - the encoded tinyVersionId
+ */
+function tinyEncode(str) {
+    const total = LENGTH_TS + LENGTH_SEQ;
+    const half = total / 2;
+    const part1 = Number(str.substring(0, half));
+    const part2 = Number(str.substring(half, total));
+    const part3 = Buffer.from(str.substring(total));
+    const enc1 = base62Integer.encode(part1);
+    const enc2 = base62Integer.encode(part2);
+    const enc3 = base62String.encode(part3);
+    return base62Integer.encode(enc1.length) +
+        enc1 +
+        base62Integer.encode(enc2.length) +
+        enc2 +
+        base62Integer.encode(enc3.length) +
+        enc3;
+}
+
+/**
+ * Decode a tinyVersionId. May return an error if the input string is
+ * invalid hex string or results in an invalid value.
+ *
+ * @param {string} str - the encoded tinyVersionId to decode
+ * @return {(string|Error)} - the decoded versionId or an error
+ */
+function tinyDecode(str) {
+    try {
+        let start = 0;
+        const enc1LenStr = str.substring(start, 1);
+        const enc1Len = base62Integer.decode(enc1LenStr);
+        if (enc1Len > 6) {
```

Comment:
This check makes me think that if you pad the encoded `enc1` with leading 0s (or the equivalent base62 character) to reach 6 chars, you don't need to embed the length, and similarly for `enc2`. Not sure of this but in case `enc1` is usually 6 chars, you can save one char.

Reply:
Indeed.